examples/apps/cli/main.c

```diff
@@ -97,3 +98,18 @@ int main(int argc, char *argv[])
 
     return 0;
 }
+
+/*
+ * Provide a "weak" log function the platform code can override if desired.
+ */
+#if !OPENTHREAD_ENABLE_DEBUG_UART_LOG
+OT_TOOL_WEAK
+void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, ...)
+{
+    va_list ap;
+
+    va_start(ap, aFormat);
+    otCliPlatLogv(aLogLevel, aLogRegion, aFormat, ap);
+    va_end(ap);
+}
```

Comment:
Hilarious - I sort of went this road and then changed direction and did it the way I did.

I like something more explicit (remove the world default)

```
/* This is effectively an ENUM so it can be #if/#else/#endif tested */
#define  OPENTHREAD_LOG_OUTPUT_NONE            0
#define OPENTHREAD_LOG_OUTPUT_DEBUG_UART 1
#define OPENTHREAD_LOG_OUTPUT_STREAM          2
/* ... future? ... */

/** Here we require the user to provide some type of log output */
#if !defined(OPENTHREAD_LOG_OUTPUT)
#define OPENTHREAD_LOG_OUTPUT    OPENTHREAD_LOG_OUTPUT_NONE
#endif
```